### PR TITLE
Add list functions for edge gateway NSXV entities

### PR DESCRIPTION
**Pull request by dataclouder**
Add new list functions needed for `vcd_resource_info`

### CODE REVIEW

File: govcd/nsxv_nat.go [hunk 1]
```diff
@@ -112,8 +106,23 @@ func (egw *EdgeGateway) GetNsxvNatRuleById(id string) (*types.EdgeNatRule, error
 	if err != nil {
 		return nil, err
 	}
+	return natRuleResponse.NatRules.EdgeNatRules, nil
+}
+
+// GetNsxvNatRuleById retrieves types.EdgeNatRule by NAT rule ID as shown in the UI using proxied
+// NSX-V API.
+// It returns and error `ErrorEntityNotFound` if the NAT rule is now found.
```
Comment (Didainius) [inline govcd/nsxv_nat.go:114]:
```suggestion
// It returns and error `ErrorEntityNotFound` if the NAT rule is not found.
```

**Review by Didainius — APPROVED** [review]

File: govcd/nsxv_nat.go (hunk 1 above)
Comment (dataclouder) [reply govcd/nsxv_nat.go]: Thanks. Fixed

File: govcd/lbappprofile_test.go [hunk 2]
```diff
@@ -71,6 +71,17 @@ func (vcd *TestVCD) Test_LBAppProfile(check *C) {
 
 	check.Assert(createdLbAppProfile.Template, Equals, lbAppProfileConfig.Template)
 
+	// Test that we can extract a list of LB app profiles, and that one of them is the profile we have got when searching by name
+	lbAppProfiles, err := edge.GetLbAppProfiles()
+	check.Assert(err, IsNil)
```
Comment (vbauzysvmware) [inline govcd/lbappprofile_test.go:76]: check that `lbAppProfiles` is not nill is missing

File: govcd/lbapprule_test.go [hunk 3]
```diff
@@ -57,6 +57,17 @@ func (vcd *TestVCD) Test_LBAppRule(check *C) {
 
 	check.Assert(createdLbAppRule.Script, Equals, lbAppRuleConfig.Script)
 
+	// Test that we can extract a list of LB app rules, and that one of them is the rule we have got when searching by name
+	lbAppRules, err := edge.GetLbAppRules()
+	check.Assert(err, IsNil)
```
Comment (vbauzysvmware) [inline govcd/lbapprule_test.go:62]: check that `lbAppRules` is not nill is missing

File: govcd/nsxv_nat_test.go [hunk 4]
```diff
@@ -131,6 +131,17 @@ func testNsxvNat(natRule *types.EdgeNatRule, vcd *TestVCD, check *C, edge EdgeGa
 
 	check.Assert(updatedNatRule.Description, Equals, natRule.Description)
 
+	// Test that we can extract a list of NSXV NAT rules, and that one of them is the rule we have got when searching by ID
+	natRules, err := edge.GetNsxvNatRules()
+	check.Assert(err, IsNil)
```
Comment (vbauzysvmware) [inline govcd/nsxv_nat_test.go:136]: check that `natRules` is not nill is missing

File: govcd/system_test.go [hunk 5]
```diff
@@ -615,3 +615,19 @@ func (vcd *TestVCD) Test_GetStorageProfileByHref(check *C) {
 	check.Assert(foundStorageProfile, Not(Equals), types.VdcStorageProfile{})
 	check.Assert(foundStorageProfile, NotNil)
 }
+
+func (vcd *TestVCD) Test_GetOrgList(check *C) {
+
+	orgs, err := vcd.client.GetOrgList()
+	check.Assert(err, IsNil)
```
Comment (vbauzysvmware) [inline govcd/system_test.go:622]: check that `orgs` is not nill is missing

**Review by vbauzysvmware — APPROVED** [review]
LGTM

File: govcd/lbapprule_test.go (hunk 3 above)
Comment (dataclouder) [reply govcd/lbapprule_test.go:62]: done

File: govcd/lbappprofile_test.go (hunk 2 above)
Comment (dataclouder) [reply govcd/lbappprofile_test.go:76]: done

File: govcd/nsxv_nat_test.go (hunk 4 above)
Comment (dataclouder) [reply govcd/nsxv_nat_test.go:136]: done

File: govcd/system_test.go (hunk 5 above)
Comment (dataclouder) [reply govcd/system_test.go:622]: done

**Review by lvirbalas — APPROVED** [review]
Please don't forget to add MaxSupportedVersion change to the commit message (it's not in PR description yet). Otherwise, LGTM!

**Pull request merged**